### DISCUSSION TITLE
closewindow unaffected by &&

### DISCUSSION THREAD

**Discussion by Kommynct** [hyprwm/Hyprland#6416] [Q&A] [answered]
I have a bind that I want to add

```
bindl = ALT, apostrophe, exec, pactl set-default-sink $(pactl list short sinks | awk '{print $2}' | sd 'easyeffects_sink\n' '' | sd "$(pactl get-default-sink)\n" '' | tofi --prompt-text " " --height 40% --width 40% --auto-accept-single true ) && pw-play --volume=0.2 /usr/share/sounds/freedesktop/stereo/audio-volume-change.oga

bindl = ALT,apostrophe,exec, hyprctl dispatch closewindow mpvrainsounds && env PIPEWIRE_NODE=alsa_output.pci-0000_06_00.hdmi-stereo mpv --wayland-app-id=mpvrainsounds --start=0 --loop --pause=no --background=color --background-color=0/0/0/.03 --player-operation-mode=pseudo-gui ~/Documents/fire.ogg

```
This would make it so that if i switch my audio sinks, my noisemaker would automatically reopen on the only sink I want it on if it was already open, however, closewindow seems to not be affected by && or ||, is there a workaround for closing a class of window/appid that actually returns whether or not it was successful?

**Comment by vaxerski** [comment, marked as answer]
nope. You probably want to `grep` `hyprctl clients`

**Reply by Kommynct** [reply]
`bindl = ALT,apostrophe,exec, hyprctl clients | rg -q 'mpvrainsounds' && hyprctl dispatch closewindow mpvrainsounds && env PIPEWIRE_NODE=alsa_output.pci-0000_06_00.hdmi-stereo mpv --wayland-app-id=mpvrainsounds --start=0 --loop --pause=no --background=color --background-color=0/0/0/.03 --player-operation-mode=pseudo-gui ~/Documents/fire.ogg`

works! thank you.
